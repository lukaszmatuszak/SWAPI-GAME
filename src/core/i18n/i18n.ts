import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import locales from './locales/index';

i18n.use(initReactI18next)
    .use(LanguageDetector)
    .init({
        ns: ['Common'],
        defaultNS: 'Common',
        resources: locales,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false,
        },
    });
