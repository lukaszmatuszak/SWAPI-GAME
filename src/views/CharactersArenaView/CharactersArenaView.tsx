import React, { useEffect } from 'react';
import { Grid, Typography, Box } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

import Battleground from '../../components/Battleground/Battleground';
import { useFetch } from '../../hooks/useFetch';
import { Character } from '../../interfaces/Character';
import { ApiUrls } from '../../core/urls/ApiUrls';
import Spinner from '../../components/Spinner/Spinner';

const CharactersArenaView: React.FC = () => {
    const [characters, isLoading] = useFetch<Character>(ApiUrls.CHARACTERS);
    const { t } = useTranslation();

    if (isLoading) {
        return <Spinner />;
    }

    return (
        <Grid container direction="column">
            <Grid item xs={12}>
                <Box pt={6} pb={6}>
                    <Typography align="center" variant="h2">
                        {t('CHARACTERS_ARENA')}
                    </Typography>
                </Box>
            </Grid>
            <Grid item xs={12}>
                <Battleground
                    contestants={characters.filter(
                        character => character.mass !== 'unknown',
                    )}
                />
            </Grid>
        </Grid>
    );
};

export default CharactersArenaView;
