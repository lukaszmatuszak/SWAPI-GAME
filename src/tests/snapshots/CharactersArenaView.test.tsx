import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CharactersArenaView from '../../views/CharactersArenaView/CharactersArenaView';
import '../../core/i18n/i18n';

describe('Characters Snapshot - ', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CharactersArenaView />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
