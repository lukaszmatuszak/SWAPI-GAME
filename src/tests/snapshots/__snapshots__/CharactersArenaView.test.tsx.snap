// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Snapshot -  renders correctly 1`] = `<Spinner />`;
